src/utils/JSONReviver.ts

```diff
@@ -8,12 +8,22 @@ type JsonableClass = (new () => { toJSON: () => IReviverValue }) & {
   validationData?: ObjectValidator<any>;
 };
 
-export interface IReviverValue<T = any> {
```

Comment:
This really gets at the heart of it. The old definition was more correct; after `isReviverValue`, we only know that it has a `ctor` and `data` field, but know *nothing* about the type of the data. The only thing wrong with the old type was that T defaulted to "any" instead of "unknown".